src/magneticField/PolarizedSingleModeMagneticField.cpp

```diff
@@ -0,0 +1,124 @@
+#include "crpropa/magneticField/PolarizedSingleModeMagneticField.h"
+
+namespace crpropa {
+
+PolarizedSingleModeMagneticField::PolarizedSingleModeMagneticField(double B_0, double wavelength, Vector3d r_0, Vector3d e_1, Vector3d e_2) {
+	setB0(B_0);
+	setWavelength(wavelength);
+	setr0(r_0);
+	sete1(e_1);
+	sete2(e_2);
+}
+
+Vector3d PolarizedSingleModeMagneticField::getField(const Vector3d &pos, const double &sigma) const {
+	Vector3d deltar = pos - r_0;
+
+	Vector3d unitVector_1;
+	if (e_1.getR() == 0)
+		throw std::runtime_error("Vector e_1 cannot be zero.");
+	unitVector_1 = e_1 / e_1.getR();
+
+	Vector3d unitVector_2;
+	if (e_2.getR() == 0)
+		throw std::runtime_error("Vector e_2 cannot be zero.");
+	unitVector_2 = e_2 / e_2.getR();
+
+	Vector3d wavevector = e_2.cross(e_1);
+	if (wavevector.getR() == 0)
+		throw std::runtime_error("e_1 cannot be parallel to e_2.");
+	wavevector = wavevector / wavevector.getR();
+
+	if (wavelength == 0)
+		throw std::runtime_error("The correlation length cannot be zero.");
+	wavevector = wavevector/wavevector.getR() * 2 * M_PI / wavelength;
+
+	return B_0 * (unitVector_1 * cos(wavevector.dot(deltar)) + unitVector_2 * sigma * sin(wavevector.dot(deltar)));
+}
+
+Vector3d PolarizedSingleModeMagneticField::getMaxHelFracField(const Vector3d &pos, const double &fH, const std::string &Bflag) const {
+	if (abs(fH) > 1)
+		throw std::runtime_error("The value of sigma has to be in the range [-1;1].");
+
+	if (Bflag == "B0") {
```

Comment:
Is B0 the most appropriate name for the flag? Will the user know that it means amplitude?
Perhaps Bflag="amplitude" and "rms" would be more adequate. (I suggest to remove the 'B' from 'Brms' because 'B' is already contained in the name 'Bflag'.

Reply:
I agree - that is a better way to do it, I will take care of that.

Reply:
In the new version I renamed the flags according to your suggestions.